server/lib/set-headers-to-preload-assets.js

```diff
@@ -32,16 +36,20 @@ function setHeadersToPreloadAssets(res, pageOptions) {
     return `<${styleUrl}>; rel=preload; as=style`;
   });
 
-  // TODO: We should preload fonts as well.
-  //
   // We use `crossorigin` because fonts are fetched with anonymous mode "cors" and
   // "same-origin" credentials mode (see
   // https://drafts.csswg.org/css-fonts/#font-fetching-requirements). `crossorigin` is
   // just short-hand for `crossorigin=anonymous` (see
   // https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/crossorigin and
   // https://html.spec.whatwg.org/multipage/infrastructure.html#cors-settings-attribute).
-  //
-  // `Link: <font_to_load.woff2>; rel=preload; as=font; crossorigin`
+  const fontLinks = fonts.map((fontUrl) => {
+    return `<${fontUrl}>; rel=preload; as=font; crossorigin`;
```

Comment:
This works to preload but we currently preload a set of fonts that isn't used becayse the Hydrogen CSS has a query parameter added to font URL in the CSS which counts as a different URL than what we preloaded.

![](https://user-images.githubusercontent.com/558581/234540176-55c2d946-5b67-4920-8e6e-c5ce573ba867.png)

Reply:
One potential solution is to use a PostCSS plugin to remove the query parameter from the CSS since we already include file hashes in the file name.

Reply:
It looks like the Hydrogen crew already ran into this and removed the query parameter from their source via https://github.com/vector-im/hydrogen-web/pull/961

So we just need to update our [Hydrogen scratch branch](https://github.com/vector-im/hydrogen-web/pull/653) with the latest changes from Hydrogen and make a release ⏩

Reply:
Tested and we only request one set of fonts now ✅

![](https://user-images.githubusercontent.com/558581/234706458-34c0b04b-658b-4d55-8532-52d2a8a5a2a2.png)